AABB_tree/examples/AABB_tree/AABB_cached_bbox_example.cpp

```diff
@@ -0,0 +1,93 @@
+#include <iostream>
+#include <fstream>
+
+#include <CGAL/Exact_predicates_inexact_constructions_kernel.h>
+#include <CGAL/AABB_tree.h>
+#include <CGAL/AABB_traits.h>
+#include <CGAL/Polyhedron_3.h>
+#include <CGAL/Surface_mesh.h>
+#include <CGAL/AABB_face_graph_triangle_primitive.h>
+#include <CGAL/Timer.h>
+
+typedef CGAL::Exact_predicates_inexact_constructions_kernel K;
+typedef K::FT FT;
+typedef K::Point_3 Point_3;
+typedef CGAL::Bbox_3 Bbox_3;
+typedef CGAL::Surface_mesh<Point_3> Surface_mesh; 
+typedef CGAL::Polyhedron_3<K> Polyhedron_3;
+typedef CGAL::Timer Timer;
+
+
+template <typename TriangleMesh>
+void triangle_mesh(char* fname)
+{
+  typedef CGAL::AABB_face_graph_triangle_primitive<TriangleMesh> Primitive;
+  typedef CGAL::AABB_traits<K, Primitive> Traits;
+  typedef CGAL::AABB_tree<Traits> Tree;
+
+  TriangleMesh tmesh;
+  std::ifstream in(fname);
+  in >> tmesh; 
+  Timer t;
+  t.start();
+  Tree tree(faces(tmesh).first, faces(tmesh).second, tmesh);
+  tree.build();
+  std::cout << t.time() << " sec." << std::endl;
+  std::cout << "Closest point to ORIGIN:" << tree.closest_point(CGAL::ORIGIN) << std::endl;
+}
+
+
+Bbox_3 bbox(boost::graph_traits<Surface_mesh>::face_descriptor fd, 
+            const Surface_mesh& p)
+{
+  boost::graph_traits<Surface_mesh>::halfedge_descriptor hd = halfedge(fd,p);
+  Bbox_3 res = p.point(source(hd,p)).bbox();
+  res += p.point(target(hd,p)).bbox();
+  res += p.point(target(next(hd,p),p)).bbox();
+  return res;
+}
+
+
+void surface_mesh_cache_bbox(char* fname)
```

Comment:
`fname` should be `const char*`, and that should solve the compilation issue.

If I have time later today, I will test that on my machine.